### PR TITLE
[Doc] Explained a bit more about the possible contents of the "name" property

### PR DESCRIPTION
**Pull request by javiereguiluz**
In the JSON schema it says that this property is of type "string" ... but maybe we can explain this a bit more and recommend to use good package names.

### CODE REVIEW

File: doc/04-schema.md [hunk 1]
```diff
@@ -29,13 +29,16 @@ The config of dependencies is ignored. This makes the `config` field
...
-separated by `/`.
-
-Examples:
+separated by `/`. Examples:
 
 * monolog/monolog
 * igorw/event-source
 
+The name can contain any character, including white spaces, and it's case
+sensitive (`foor/bar` is considered a different package than `Foo/Bar`). In order
```
Comment (Seldaek) [inline doc/04-schema.md]: That is incorrect :) Package names are case-insensitive. We just enforce lowercase on packagist for uniformity's sake.

Comment (javiereguiluz) [inline doc/04-schema.md]: Fixed! Thanks.

File: doc/04-schema.md [hunk 2]
```diff
@@ -29,13 +29,16 @@ The config of dependencies is ignored. This makes the `config` field
...
-separated by `/`.
-
-Examples:
+separated by `/`. Examples:
 
 * monolog/monolog
 * igorw/event-source
 
+The name can contain any character, including white spaces, and it's case
+insensitive (`foor/bar` and `Foo/Bar` are considered the same package). In order
```
Comment (stloyd) [inline doc/04-schema.md]: `foor/bar` is not same as `For/Bar` (`foo/bar` in fact) ;)

**Comment by Seldaek** [comment]
Thanks

File: doc/04-schema.md [hunk 3]
```diff
@@ -29,13 +29,16 @@ The config of dependencies is ignored. This makes the `config` field
...
-Examples:
+separated by `/`. Examples:
 
 * monolog/monolog
 * igorw/event-source
 
+The name can contain any character, including white spaces, and it's case
+insensitive (`foo/bar` and `Foo/Bar` are considered the same package). In order
+to simplify its installation, it's recommended to define a short and lowercase
+name that doesn't include non-alphanumeric characters or white spaces.
```
Comment (stof) [inline doc/04-schema.md]: shared packages have some extra requirements (due to the fact that the name will be used in a filesystem path). And Packagist enforces the strict regex.

**Pull request merged**